stack.yaml

```diff
@@ -0,0 +1,5 @@
+
+resolver: nightly-2024-07-22
+
+packages:
+- ./.
```

Comment:
I might suggest keeping this PR isolated to making the required changes for GHC 9.8 and supporting the new `base64` API.

I would raise the `stack.yaml` changes separately.